apps/web/src/studio/components/workflows/test-workflow/WorkflowsTestPage.tsx

```diff
@@ -0,0 +1,100 @@
+import { useDisclosure } from '@mantine/hooks';
+import { Button } from '@novu/novui';
+import { IconOutlineCable, IconPlayArrow } from '@novu/novui/icons';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { bridgeApi } from '../../../../api/bridge/bridge.api';
+import { testTrigger } from '../../../../api/notification-templates';
+import { When } from '../../../../components/utils/When';
+import { useAuth } from '../../../../hooks/useAuth';
+import { ExecutionDetailsModalWrapper } from '../../../../pages/templates/components/ExecutionDetailsModalWrapper';
+import { WorkflowsPageTemplate, WorkflowsPanelLayout } from '../layout/index';
+import { ToSubscriber, WorkflowTestInputsPanel } from './WorkflowTestInputsPanel';
+import { WorkflowTestTriggerPanel } from './WorkflowTestTriggerPanel';
+
+export const WorkflowsTestPage = () => {
+  const { currentUser, isLoading: isAuthLoading } = useAuth();
+  const { templateId = '' } = useParams<{ templateId: string }>();
+  const [payload, setPayload] = useState<Record<string, any>>({});
+  const [to, setTo] = useState<ToSubscriber>({
+    subscriberId: '',
+    email: '',
+  });
+
+  const { data: workflow, isLoading: isWorkflowLoading } = useQuery(['workflow', templateId], async () => {
+    return bridgeApi.getWorkflow(templateId);
+  });
+
+  const stepTypes = useMemo(() => {
+    if (!workflow) {
+      return [];
+    }
+
+    return workflow.steps.map((step) => step.type);
+  }, [workflow]);
+
+  const { mutateAsync: triggerTestEvent, isLoading: isTestLoading } = useMutation(testTrigger);
+  const [transactionId, setTransactionId] = useState<string>('');
+  const [executionModalOpened, { close: closeExecutionModal, open: openExecutionModal }] = useDisclosure(false);
+
+  const handleTestClick = async () => {
+    const response = await triggerTestEvent({
+      name: workflow.workflowId,
+      to,
+      payload: {
+        ...payload,
+        __source: 'studio-test-workflow',
+      },
+    });
+
+    setTransactionId(response.transactionId || '');
+    openExecutionModal();
+  };
+
+  const onChange = (payloadValues, toValues) => {
+    if (toValues) {
+      setTo(toValues);
+    }
+    if (payloadValues) {
+      setPayload(payloadValues);
+    }
+  };
+
+  if (isAuthLoading || isWorkflowLoading || isTestLoading || !currentUser) {
```

Comment:
`isAuthLoading` ensures that user and org is loaded. 

```suggestion
  if (isAuthLoading || isWorkflowLoading || isTestLoading) {
```